src/LSAG.hs

```diff
@@ -204,6 +207,25 @@ pointToBS (ECC.Point x y) = show x <> show y
 pubKeysToBS :: [ECDSA.PublicKey] -> BS.ByteString
 pubKeysToBS = foldMap (pointToBS . ECDSA.public_q)
 
+
+-- | Iterative algorithm to generate H.
+-- The important to hide its discrete logarithm "k" such that H = kG
+generateH :: ECC.Point -> ECC.Curve -> [Char] -> ECC.Point
+generateH g curve extra =
+  case yM of
+    Nothing -> generateH g curve (toS $ '1':extra)
```

Comment:
The use of this magic `'1'` character is a little odd. What's the context on this choice?

Reply:
`x` is derived from the hash. The algorithm needs to iterate to generate a new `x` when there isn't a `y` s.t. `y = x^3 + 7 mod p`. We need to deterministically create a new hash from the existing hash. `1` is an arbitrary change to the existing hash

Reply:
Ok, I see. This makes sense.

Reply:
Do you have any suggestions to make it more sensible?

Reply:
Added a name to the magic number